pinot-common/src/main/java/org/apache/pinot/common/utils/config/TableConfigUtils.java

```diff
@@ -251,7 +251,7 @@ public static void convertFromLegacyTableConfig(TableConfig tableConfig) {
     if (batchIngestionConfig == null) {
       // Only create the config if any of the deprecated config is not null.
       if (segmentPushType != null || segmentPushFrequency != null) {
-        batchIngestionConfig = new BatchIngestionConfig(null, segmentPushType, segmentPushFrequency);
+        batchIngestionConfig = new BatchIngestionConfig(null, segmentPushType, segmentPushFrequency, false, false);
```

Comment:
I tried to overload `BatchIngestionConfig.java` in order to not have to add the two additional parameters `consistentDataPush` and `disableUniqueSegmentNames` everywhere, however, seems like `@JsonCreator` annotation on `BatchIngestionConfig`'s constructor enforces a unique constructor. 

I'm guessing one way around this is to introduce `BatchIngestionConfigBuilder`?